src/System.Collections.Specialized/src/System/Collections/Specialized/NameValueCollection.cs

```diff
@@ -144,7 +147,8 @@ protected void InvalidateCachedArrays()
                 return null;
 
             string[] array = new string[n];
-            list!.CopyTo(0, array, 0, n);
+            Debug.Assert(list != null);
+            list.CopyTo(0, array, 0, n);
```

Comment:
I would just use a `!` here instead of an assert.  It's obvious from the control flow in the short method that `list` will not be null here; the compiler just can't see it.